src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/OneDayFileJsonParser.java

```diff
@@ -40,6 +47,18 @@ public MetricsInfoData parseOneDayFile (File inputFile) {
                 // converts JSON to Map
                 jsonMap = objectMapper.readValue(fileLine, new TypeReference<Map<String, Object>>(){});
 
+                // get Origins & Destinations country code
+                int originCountryCode =-1;
+                int destinationCountryCode =-1;
+                if ( !("".equals(jsonMap.get("origin"))) ) {
+                    originCountryCode = extractCountryCodeFromMsisdn(
+                            ((Number)jsonMap.get("origin")).longValue() );
```

Comment:
V 99% případů, pokud se Ti někde opakují stejné kusy kódu, tak si ta logika říká o to, aby byla o kus jinde.
Třeba tady by se dalo krásně přesunout (Number) ... longValue() for metody extractCountryCodeFromMsisdn(jsonMap.get("origin"))

Reply:
předěláno vč. nové beany CountryCodeExtractor -> je na něj delegována ta vlastní operace extrakce CC